src/kernel/src/hv/ram.rs

```diff
@@ -40,6 +39,11 @@ impl Ram {
             mem.cast()
         };
 
+        Ok(Self { addr, mem })
+    }
+
+    #[cfg(windows)]
+    pub fn new(addr: usize) -> Result<Self, Error> {
```

Comment:
The downside with splitting the public method like this is we will have a duplicated documentation so better to leave it merged like the previous version. If you want to split it better to split it into something like `new_unix` and `new_win32` as private functions and call it from `new` instead.